src/boot/efi/stub.c

```diff
@@ -104,6 +105,27 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
         if (disk_get_part_uuid(loaded_image->DeviceHandle, uuid) == EFI_SUCCESS)
                 efivar_set(L"LoaderDevicePartUUID", uuid, FALSE);
 
+        /* if LoaderImageIdentifier not set, assume the image with this stub was loaded directly from UEFI */
+        if (efivar_get_raw(&global_guid, L"LoaderImageIdentifier", &b, &size) != EFI_SUCCESS) {
+                CHAR16 *loaded_image_path = DevicePathToStr(loaded_image->FilePath);
+                efivar_set(L"LoaderImageIdentifier", loaded_image_path, FALSE);
+        }
+
+        /* same as above with LoaderInfo */
+        if (efivar_get_raw(&global_guid, L"LoaderInfo", &b, &size) != EFI_SUCCESS)
+                efivar_set(L"LoaderInfo", L"systemd-stub " PACKAGE_VERSION, FALSE);
+
+        if (efivar_get_raw(&global_guid, L"LoaderFirmwareInfo", &b, &size) != EFI_SUCCESS) {
+                s = PoolPrint(L"%s %d.%02d", ST->FirmwareVendor, ST->FirmwareRevision >> 16, ST->FirmwareRevision & 0xffff);
```

Comment:
I do not know well about `PoolPrint()`, but should we check allocation failure here?

Reply:
All the other uses of `PoolPrint()` in systemd don't check for allocation failure